projects/exui-common-lib/src/lib/components/search-location/search-location.component.html

```diff
@@ -1,13 +1,14 @@
 <div class="auto-complete-container">
-  <div class="autocomplete__wrapper">                
-    <input                             
+  <div class="autocomplete__wrapper">
+    <input
       [formControl]="findLocationFormGroup.controls.findLocationFormControl"
       [matAutocomplete]="auto"
+      (keydown)="onKeyDown()"
       (keyup)="onKeyUp($event)"
```

Comment:
If you look at the methods you will understand why  or @paulhowes-HMCTS we could explain in Booking @rayliangatos

Reply:
@rayliangatos @waynepHMCTS I have thoroughly tested this and we can't merge the KeyDown and Keyup events. You need a KeyDown method to prevent the user from deleting a location then moving away from the input without calling the KeyUp event as this forces form validation. KeyUp is used to filter the Locations and if this is put in the KeyDown event the search uses the last input field value not the latest! Merging into either the KeyDown or KeyUp causes either the form validation or the search filter to fail.

Reply:
thanks both clarify this